### PR TITLE
fix clippy lints

### PR DESCRIPTION
**Pull request by robjtede**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #![deny(rust_2018_idioms)]
 #![warn(missing_debug_implementations)]
+#![allow(clippy::type_complexity)]
```
Comment (Enet4) [inline src/main.rs:3]: The seven warnings coming from this lint are mostly of the form `Pin<Box<dyn Future<Output = Result<T, E>> + Send>`. Does it go away if we replace them with some sort of `PinnedResultFuture<T, E>` alias?

**Review by Enet4 — COMMENTED** [review]
This does make the code look better in general!

File: src/main.rs (hunk 1 above)
Comment (paolobarbolini) [reply src/main.rs:3]: Something like [futures::future::BoxFuture](https://docs.rs/futures/0.3.5/futures/future/type.BoxFuture.html)

**Review by Enet4 — APPROVED** [review]

**Review by Feliix42 — APPROVED** [review]
Looks good to me as well!

**Pull request merged**